shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Snap.hs

```diff
@@ -44,28 +49,41 @@ data SnapPredicateFailure era -- No predicate failures
 instance NoThunks (SnapPredicateFailure era)
 
 instance (UsesTxOut era, UsesValue era) => STS (SNAP era) where
-  type State (SNAP era) = SnapShots (Crypto era)
-  type Signal (SNAP era) = ()
+  type State (SNAP era) = SnapShots era
+  type Signal (SNAP era) = SlotNo
```

Comment:
It looks like we actually only need `EpochNo` rather than `SlotNo` here. I'd feel better having the epoch number as the argument to `NEWEPOCH`, `EPOCH` etc to make it clear we don't do anything that's conditional on the actual slot that ticks us into that epoch.

Reply:
Actually, after studying your comment, I realized we don't have to change the Environment of the NEWEPOCH or of the EPOCH rules at all. It can remain (). This makes things much simpler, so thank for the observation!